### PR TITLE
add grpc example

### PR DESCRIPTION
**Pull request by thinkerou**
use grpc helloworld example.

### CODE REVIEW

File: examples/grpc/gin/main.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	conn, err := grpc.Dial("localhost:50051", grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+	c := pb.NewGreeterClient(conn)
+
+	// Set up a http setver.
+	r := gin.Default()
+	r.GET("/rest/n/:name", func(g *gin.Context) {
```
Comment (appleboy) [inline examples/grpc/gin/main.go]: maybe change `g *gin.Context` to `c *gin.Context`.

File: vendor/vendor.json [hunk 2]
```diff
@@ -114,6 +114,12 @@
 			"revision": "99f16d856c9836c42d24e7ab64ea72916925fa97",
 			"revisionTime": "2017-03-08T15:04:45Z"
 		},
+		{
+			"checksumSHA1": "alYroN0n+3NDlMXZeiC6d/LG764=",
+			"path": "google.golang.org/grpc",
+			"revision": "9cb02b885b41e0db982e23d6e51a24ea465dec2e",
+			"revisionTime": "2017-06-26T23:20:44Z"
+		},
```
Comment (appleboy) [inline vendor/vendor.json]: Please don't update the vendor source code from example code.

Comment (thinkerou) [reply vendor/vendor.json]: got it, but if have not this, ci will fail.

File: examples/grpc/gin/main.go (hunk 1 above)
Comment (thinkerou) [reply examples/grpc/gin/main.go]: OK, thanks.

File: vendor/vendor.json (hunk 2 above)
Comment (thinkerou) [reply vendor/vendor.json]: and I have removed it.

**Comment by thinkerou** [comment]
```sh
examples/grpc/pb/helloworld.pb.go:23:2: cannot find package "google.golang.org/grpc" in any of:
	/home/travis/gopath/src/github.com/gin-gonic/gin/vendor/google.golang.org/grpc (vendor tree)
	/home/travis/.gimme/versions/go1.6.4.linux.amd64/src/google.golang.org/grpc (from $GOROOT)
	/home/travis/gopath/src/google.golang.org/grpc (from $GOPATH)
make: *** [test] Error 1
```

**Comment by appleboy** [comment]
@thinkerou ignore check `examples` folder for `go vet` command. Done, see the latest commit.

File: examples/grpc/gin/main.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		res, err := client.SayHello(g, req)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+		} else {
+			c.JSON(http.StatusOK, gin.H{
+				"result": fmt.Sprint(res.Message),
+			})
+		}
```
Comment (appleboy) [inline examples/grpc/gin/main.go]: maybe updated as the following for readable.

```go
		if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{
				"error": err.Error(),
			})
			return
		}
		
		c.JSON(http.StatusOK, gin.H{
			"result": fmt.Sprint(res.Message),
		})
```

File: examples/grpc/gin/main.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	r.GET("/rest/n/:name", func(c *gin.Context) {
+		name := c.Param("name")
+
+		// Contact the server and print out its response.
+		req := &pb.HelloRequest{Name: name}
+		res, err := client.SayHello(g, req)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
```
Comment (appleboy) [inline examples/grpc/gin/main.go]: missing `return`.

Comment (thinkerou) [reply examples/grpc/gin/main.go]: Oh, thanks a lot, fixed.

**Comment by thinkerou** [comment]
@appleboy thank you update `Makefile`, but I remember `vendor.json` have some packages from `example` folder.

**Comment by appleboy** [comment]
@thinkerou Can you also help to remove packages from example?

**Comment by thinkerou** [comment]
@appleboy OK, I will alone commit one pull request for coding review.

**Comment by appleboy** [comment]
build error.

**Comment by thinkerou** [comment]
code indent error, fixed.

**Pull request merged**